src/test/ui/error-codes/E0746.stderr

```diff
@@ -0,0 +1,27 @@
+error[E0746]: return type cannot have a bare trait because it must be `Sized`
+  --> $DIR/E0746.rs:8:13
+   |
+LL | fn foo() -> dyn Trait { Struct }
+   |             ^^^^^^^^^ doesn't have a size known at compile-time
+   |
+   = note: for information on `impl Trait`, see <https://doc.rust-lang.org/book/ch10-02-traits.html#returning-types-that-implement-traits>
+help: you can use the `impl Trait` feature in the return type because all the return paths are of type `Struct`, which implements `dyn Trait`
```

Comment:
"implements \`dyn Trait\`" seems wrong -- `dyn Trait` is a type, not a trait, and types don't implement other types. The dyn keyword was created to distinguish dyn Trait the type from Trait the trait.